mail/views.py

```diff
@@ -0,0 +1,102 @@
+"""Mail views"""
+from types import SimpleNamespace
+
+from django.conf import settings
+from django.http import JsonResponse
+from django.shortcuts import render
+from django.utils.decorators import method_decorator
+from django.views import View
+from django.views.decorators.csrf import csrf_exempt
+
+from mail import api
+from mail.forms import EmailDebuggerForm
+from sites.api import get_default_site
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class EmailDebuggerView(View):
+    """Email debugger view"""
+    form_cls = EmailDebuggerForm
+    initial = {}
+    template_name = 'email_debugger.html'
+
+    def get(self, request):
+        """
+        Dispalys the debugger UI
+        """
+        form = self.form_cls(initial=self.initial)
+        return render(request, self.template_name, {
+            'form': form,
+        })
+
+    def post(self, request):
+        """
+        Renders a test email
+        """
+        form = self.form_cls(request.POST)
+
+        if not form.is_valid():
+            return JsonResponse({"error": "invalid input"})
+
+        email_type = form.cleaned_data['email_type']
+        context = {
+            'base_url': settings.SITE_BASE_URL,
+            'anon_token': 'abc123',
+            'site_name': get_default_site().name,
```

Comment:
when I try to render a test email I get an error on this line, saying that `AuthenticatedSite` has no attribute name - do I have something misconfigured?

Reply:
I usually solve that by adding an AuthenticatedSite through the django admin

Reply:
yeah, I already had one, but the `AuthenticatedSite` model doesn't have a `name` attribute. I think Nathan fixed the issue.